src/ripple/overlay/impl/PeerImp.h

```diff
@@ -603,6 +606,22 @@ class PeerImp : public Peer,
 
     void
     getLedger(std::shared_ptr<protocol::TMGetLedger> const& packet);
+
+    compression::Compressed
+    enableCompression()
```

Comment:
I'd prefer if we don't put implementations in the class definition. It's easier to see interfaces without them.

I would not make `enableCompression` and `enableVPReduceRelay` members. Consider putting them in the cpp file as free functions, and pass the needed info to them as parameters. Part of the motivation for that is these functions are only used in the constructor, and they depend on `headers_` being initialized before they can be used. By passing the parameter explicitly it makes this dependency clear.